src/mpid/ch4/netmod/ofi/util.c

```diff
@@ -87,7 +88,7 @@ uint64_t MPIDI_OFI_mr_key_alloc(int key_type, uint64_t requested_key)
                         mr_key_allocator.last_free_mr_key = i;
                         ret_key = i * sizeof(uint64_t) * 8 + (nval - 1);
                         /* assert local key does not exceed its range */
-                        MPIR_Assert((ret_key & (1ULL << MPIDI_OFI_MR_KEY_PREFIX_SHIFT)) == 0);
+                        MPIR_Assert((ret_key & (1ULL << key_prefix_shift)) == 0);
```

Comment:
Can you change the assert into an error return? Assertions are for catching potential coding errors. Runtime errors should use error returns.

Reply:
I didn't understand this entirely. 

Since this function returns a key, what can I return to be an erroneous key and wouldn't the calling code need to handle that?

Reply:
I guess you can return `MPIDI_OFI_INVALID_MR_KEY`, and have the caller to check for it and return error at caller sites.

Reply:
With 64 bit key space, I don't expect we'll ever exhaust the key space, so Assertion is fine. But with 2-byte key size, exhaustion can happen, and we need return error so user potentially can handle it.

Reply:
How about `**ofid_mr_key`? You can add it to `src/mpid/ch4/netmod/ofi/errnames.txt`.